spec/integration/aws_route53_hosted_zone_spec.rb

```diff
@@ -130,6 +130,36 @@
             # the empty {} acts as a wildcard, and all zones have SOA and NS records we want to skip.
           end
 
+          it "creates a hosted zone with a RecordSet with an RR name with a trailing dot" do
+            expect_recipe {
+              aws_route53_hosted_zone "feegle.com" do
+                record_sets {
+                  aws_route53_record_set "some-host.feegle.com." do
+                    type "CNAME"
+                    ttl 3600
+                    resource_records ["some-other-host"]
+                  end
+                }
+              end
+            }.to create_an_aws_route53_hosted_zone("feegle.com",
+                                                   resource_record_sets: [{}, {}, sdk_cname_rr]).and be_idempotent
+          end
+
+          # AWS's error for this is "FATAL problem: DomainLabelEmpty encountered", so we help the user out.
```

Comment:
Nice improvement on a bad error, and thanks for the test that answered a question I had